File: src/main/java/de/danielmaile/aether/worldgen/dungeon/Direction.java
package de.danielmaile.aether.worldgen.dungeon;

import org.bukkit.util.Vector;

public enum Direction
{
    EAST(new Vector(1, 0, 0)),
    WEST(new Vector(-1, 0, 0)),
    SOUTH(new Vector(0, 0, 1)),
    NORTH(new Vector(0, 0, -1));

    public Vector getRelativePos()
    {
        return relativePos;
    }

    private final Vector relativePos;

    Direction(Vector relativePos)
    {
        this.relativePos = relativePos;
    }
}

File: src/main/java/de/danielmaile/aether/worldgen/dungeon/Direction.kt
package de.danielmaile.aether.worldgen.dungeon

import org.bukkit.util.Vector

enum class Direction(val relativePos: Vector) {
    EAST(Vector(1, 0, 0)),
    WEST(Vector(-1, 0, 0)),
    SOUTH(Vector(0, 0, 1)),
    NORTH(Vector(0, 0, -1));
}